R/fit_drc_4p.R

```diff
@@ -206,17 +247,16 @@ fit_drc_4p <- function(data,
     vector_add <- c(vector[1] + 1, vector[2] - 1)
     vector_add_rev <- rev(vector_add)
 
-    concentrations <- sort(unique(dplyr::pull(data_prep, {{ dose }})))
-
     filter_completeness <- data_prep %>%
       dplyr::group_by({{ grouping }}, {{ dose }}) %>%
       dplyr::mutate(enough_replicates = sum(!is.na({{ response }}))
-      >= n_replicates_completeness) %>%
+      >= n_replicate_completeness) %>%
+      dplyr::group_by({{ grouping }}, .data$enough_replicates) %>%
+      dplyr::mutate(n_condition_enough = n_distinct({{ dose }})) %>%
       dplyr::group_by({{ grouping }}) %>%
-      dplyr::mutate(enough_conditions = sum(.data$enough_replicates) /
-        n_replicates
-      >= n_conditions_completeness) %>%
-      dplyr::mutate(anova_significant = {{ grouping }} %in% anova_filtered) %>%
+      dplyr::mutate(enough_conditions = any(ifelse(.data$n_condition_enough >= n_condition_completeness &
+        .data$enough_replicates, TRUE, FALSE))) %>%
+      dplyr::select(-"n_condition_enough") %>%
```

Comment:
I think you should use this function without `any()`. even if there is only one `enough_replicate` == TRUE, the whole column will be marked as TRUE, as any() is looking at the whole column 

here a small example with the iris dataset:
![Screenshot 2024-03-11 at 15 05 24](https://github.com/jpquast/protti/assets/70535771/2ca137b4-1d69-40b9-b80b-3959817bf16a)

Reply:
It is grouped before by the `grouping` column. Therefore it will be all `TRUE` for each group if any of the rows is `TRUE`. That is the intention. I can, however, actually leave out the `ifelse()` because the results are either `TRUE` or `FALSE`, which is also the result of the logical statement.

```
 filter_completeness <- data_prep %>%
      dplyr::group_by({{ grouping }}, {{ dose }}) %>%
      dplyr::mutate(enough_replicates = sum(!is.na({{ response }}))
      >= n_replicate_completeness) %>%
      dplyr::group_by({{ grouping }}, .data$enough_replicates) %>%
      dplyr::mutate(n_condition_enough = n_distinct({{ dose }})) %>%
      dplyr::group_by({{ grouping }}) %>%
      dplyr::mutate(enough_conditions = any(.data$n_condition_enough >= n_condition_completeness &
        .data$enough_replicates))
```

What I essentially do is to first check for each concentration (`dose`) if there are enough replicates. This is in the column `enough_replicates` as a logical value. You can see 2 do not have enough replicates while 6 do. The `n_condition_enough` column contains the actual numbers. So in the test for enough conditions I then check if there are any cases in which the number of complete conditions (`n_condition_enough` + `enough_replicates == TRUE`) is actually higher than the user provided value.

<img width="898" alt="Screenshot 2024-03-17 at 16 57 21" src="https://github.com/jpquast/protti/assets/59307482/57b4dbdd-e64d-41fc-bdba-11cc0a7db7a4">

Reply:
okidoki - thanks for the explanation!